tests/various/hierarchy_defer.ys

```diff
@@ -0,0 +1,19 @@
+read -vlog2k <<EOT
+module first;
+endmodule
+
+(* top *)
+module top(input i, output o);
+sub s0(i, o);
+endmodule
+
+(* constant_expression=1+1?2*2:3/3 *)
+module sub(input i, output o);
+assign o = ~i;
+endmodule
+EOT
+
+hierarchy -auto-top
```

Comment:
`hierarchy -auto-top` is meant for cases where there is no dedicated top module (marked with the `top` attribute). That's the opposite of what you are testing here. So it should just be `hierarchy` without `-auto-top`.

Reply:
The reason I am particularly interested with `-auto-top` is that all the `synth_*` commands call `hierarchy` with this option when there is no `-top`.
It is also my understanding that `-auto-top` is only treated as a fall-back when `(* top *)` cannot be find.